src/main/java/seedu/address/model/ScheduleManager.java

```diff
@@ -0,0 +1,214 @@
+        int intervalStart = 0;
+
+        for (AnnotatedPointOfTime point : queue) {
+            switch (point.getType()) {
+            case TimeIntervalStart:
+                if (!isAppointment) {
+                    intervalStart = point.getTimeValue();
+                }
+                isNoneOverlapping = true;
+                break;
+
+            case TimeIntervalEnd:
+                if (!isAppointment) {
+                    nonOverLappingIntervals.add(new TimeInterval(
+                        TimeConversionUtil.convertIntToTime(intervalStart),
+                        TimeConversionUtil.convertIntToTime(point.getTimeValue())));
+                }
+                isNoneOverlapping = false;
+                break;
+
+            case AppointmentStart:
+                if (isNoneOverlapping) {
+                    nonOverLappingIntervals.add(new TimeInterval(
+                        TimeConversionUtil.convertIntToTime(intervalStart),
+                        TimeConversionUtil.convertIntToTime(point.getTimeValue())));
+                }
+                isAppointment = true;
+                break;
+
+            case AppointmentEnd:
+                if (isNoneOverlapping) {
+                    intervalStart = point.getTimeValue();
+                }
+                isAppointment = false;
+                break;
+
+            default:
+                break;
+            }
+        }
+        return new TimeIntervalList(nonOverLappingIntervals);
+    }
+
+    /**
+     * Finds the available time slots based on the appointment list of a specified date.
+     *
+     * @return a string message of available time intervals.
+     */
+    public String findFreeSlots() {
+        TimeIntervalList operationTimeIntervals = constructOperationTimeIntervals();
+        TimeIntervalList appointmentTimeIntervals = constructAppointmentTimeInterval();
+
+        annotateOperationTimeInterval(operationTimeIntervals);
+        annotateAppointmentTime(appointmentTimeIntervals);
+
+        queueAnnotatedPoints();
+        availableTimeSlots = findNonOverlappingIntervals(annotatedPointList);
+        return availableTimeSlots.clearZeroIntervals().toString();
+    }
+
+    /**
+     * Finds the next available time slot for 1 hour appointment on a specified date.
+     *
+     * @return a string message of the next available time.
+     */
+    public String getNextAvailableSlot(TimeIntervalList availableTimeSlots) {
+        for (TimeInterval timeInterval : availableTimeSlots.getTimeIntervals()) {
+            if (timeInterval.isAtLeastOneHour()) {
+                Time startTime = timeInterval.getStartTime();
+                Time endTime = new Time(startTime.getTime().plusHours(1));
+                TimeInterval nextInterval = new TimeInterval(startTime, endTime);
+                return nextInterval.toString();
+            }
+        }
+        return "Sorry we cannot find any slot available for 1 hour appointment.";
+    }
```

Comment:
Sorry just a bit curious about this function, this is currently not used right?

Reply:
Yes it was not used at this stage but I have implemented it in the next commit.